server/src/main/java/org/opensearch/rest/RestHandler.java

```diff
@@ -163,6 +164,14 @@ public ReplacedRoute(Method method, String path, Method deprecatedMethod, String
             this.deprecatedPath = deprecatedPath;
         }
 
+        public ReplacedRoute(Route route, String prefix, String deprecatedPrefix) {
+            this(route.getMethod(), prefix + route.getPath(), route.getMethod(), deprecatedPrefix + route.getPath());
```

Comment:
```suggestion
            this(route.getMethod(), prefix + route.getPath(), deprecatedPrefix + route.getPath());
```
use newly introduced ctor

Reply:
Done